libcanard/canard.c

```diff
@@ -401,14 +401,20 @@ CANARD_PRIVATE int32_t txPushSingleFrame(struct CanardTxQueue* const        que,
         *(frame_bytes + (frame_payload_size - 1U)) = txMakeTailByte(true, true, true, transfer_id);
 
         // Insert the newly created TX item into the priority queue.
-        const struct CanardTreeNode* const priority_queue_res =
-            cavlSearch(&que->priority_root, &tqi->priority_base, &txAVLPriorityPredicate, &avlTrivialFactory);
+        const struct CanardTreeNode* const priority_queue_res = cavl2_find_or_insert(&que->priority_root,
+                                                                                     &tqi->priority_base,
+                                                                                     &txAVLPriorityPredicate,
+                                                                                     &tqi->priority_base,
+                                                                                     &avlTrivialFactory);
```

Comment:
Please remove the trivial factory from this codebase and use one provided by cavl2.